docker-compose.yml

```diff
@@ -29,12 +29,10 @@ services:
        - "11211:11211"
 
   mongo:
-    # We use WiredTiger in all environments. In development environments we use small files
-    # to conserve disk space, and disable the journal for a minor performance gain.
-    # See https://docs.mongodb.com/v3.0/reference/program/mongod/#options for complete details.
-    command: mongod --smallfiles --nojournal --storageEngine wiredTiger
+    # We disable the journal for a minor performance gain.
+    command: mongod --nojournal
     container_name: edx.devstack.mongo
-    image: mongo:4.0.24
+    image: mongo:4.2.17
```

Comment:
@umar221b Is the plan to also upgrade mongo version on staging and production to 4.2.17? If not I prefer to keep the current version as it is or at least upgrade it to 4.0.27 to match the production mongo DB version as part of mimicking the production infrastructure to minimize WOMM errors.

Upgrading to mongo version 5 is right now on Sitech Todo list as mongo 4 will not be supported by Atlas by April 2022

Reply:
I wanted to talk to you about this. I was trying to initiate the upgrade to at least 4.2 before we reach the end of life on April, the upgrade to 4.2 on Atlas should also be seamless, so I thought we can test it locally for a while (before merging) and make sure everything is looking good.

Reply:
Can you check one thing before moving forward, try to upgrade to 5 and see if it will break things or not, rather than do it twice upgrade to 4.2 then 5

Reply:
I do not think it is possible. Upgrading to 4.4 will fail because the data is not compatible with 4.4. As far as I understand we need to do this gradually.